File: 2022-2023/sourcecode/lab/lab11/FoodyJava/app/src/main/java/pl/udu/uwr/pum/foodyjava/adapters/favorite/FavoriteAdapter.java
package pl.udu.uwr.pum.foodyjava.adapters.favorite;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;

import pl.udu.uwr.pum.foodyjava.adapters.MealComparator;
import pl.udu.uwr.pum.foodyjava.data.Meal;
import pl.udu.uwr.pum.foodyjava.databinding.ListItemRvBinding;

public class FavoriteAdapter extends ListAdapter<Meal, FavoriteViewHolder> {
    public FavoriteAdapter(MealComparator comparator) {
        super(comparator);
    }

    @NonNull
    @Override
    public FavoriteViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new FavoriteViewHolder(ListItemRvBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        ));
    }

    @Override
    public void onBindViewHolder(@NonNull FavoriteViewHolder holder, int position) {
        Meal item = getItem(position);
        holder.bind(item);
    }

    public Meal getItemAt(int position){
        return getItem(position);
    }
}

File: 2022-2023/sourcecode/lab/lab11/Foody/app/src/main/java/pl/udu/uwr/pum/foody/adapters/favorite/FavoriteAdapter.kt
package pl.udu.uwr.pum.foody.adapters.favorite

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import pl.udu.uwr.pum.foody.adapters.FoodComparator
import pl.udu.uwr.pum.foody.data.Meal
import pl.udu.uwr.pum.foody.databinding.ListItemRvBinding

class FavoriteAdapter(itemComparator: FoodComparator) : ListAdapter<Meal, FavoriteViewHolder>(itemComparator) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FavoriteViewHolder {
        return FavoriteViewHolder(
            ListItemRvBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: FavoriteViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }

    fun getItemAt(position: Int): Meal{
        return getItem(position)
    }
}